### PR TITLE
Vertical scrolling and backend integration in Feed Tile

### PR DESCRIPTION
**Pull request by lukasz-ciskowski**
Added fetching data from backend and added vertical scrolling

### CODE REVIEW

File: src/tiles/FeedTile.vue [hunk 1]
```diff
@@ -1,15 +1,71 @@
...
+                    <div class="feed--author">{{ feed.author.username }}</div>
+                    <div class="feed--publishedAt">{{ feed.publishedAt.toLocaleString() }}</div>
+                </div>
+            </div>
         </div>
     </article>
 </template>
 
 <script>
+    var marquee = require('../marquee')
```
Comment (JakubBizewski) [inline src/tiles/FeedTile.vue]: It's best to avoid mixing standards. We're using `import` from ES6 and I think you should stick with it.

**Review by JakubBizewski — CHANGES_REQUESTED** [review]

File: src/marquee.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+var marqueeSpeed=1
+var actualHeight=0
+
+function Init(containerId, content)
```
Comment (Albert221) [inline src/marquee.js]: - Maybe let's name it `Marquee`?
- `export default function Marquee(...)`

File: src/marquee.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+var marqueeSpeed=1
+var actualHeight=0
```
Comment (Albert221) [inline src/marquee.js]: Don't use `var`s, use `let` or `const`

File: src/marquee.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+function Init(containerId, content)
+{
+	var crossMarquee=document.getElementById(content)
```
Comment (Albert221) [inline src/marquee.js]: Where is any code style? Spaces surrounding operators!

File: src/marquee.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+	var marqueeheight=document.getElementById(containerId).offsetHeight
+	actualHeight=crossMarquee.offsetHeight
+	
+	setInterval(function(){
```
Comment (Albert221) [inline src/marquee.js]: Space before opening bracket

File: src/marquee.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+		var crossMarquee=document.getElementById(content)
+		actualHeight=crossMarquee.offsetHeight
+
+		if (parseInt(crossMarquee.style.top)>(actualHeight * (-1) + 8)) 
+		crossMarquee.style.top=parseInt(crossMarquee.style.top)-marqueeSpeed+"px" 
+		else 
+		crossMarquee.style.top=parseInt(marqueeheight)+8+"px"
```
Comment (Albert221) [inline src/marquee.js]: AFAIR we use single quotes for this. @DarkMuffinJoe can relate?

File: src/marquee.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+	var crossMarquee=document.getElementById(content)
+	var marqueeheight=document.getElementById(containerId).offsetHeight
+	actualHeight=crossMarquee.offsetHeight
+	
+	setInterval(function(){
+		var crossMarquee=document.getElementById(content)
+		actualHeight=crossMarquee.offsetHeight
+
+		if (parseInt(crossMarquee.style.top)>(actualHeight * (-1) + 8)) 
+		crossMarquee.style.top=parseInt(crossMarquee.style.top)-marqueeSpeed+"px"
```
Comment (Albert221) [inline src/marquee.js]: You forgot indentation here and below

File: src/marquee.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+		actualHeight=crossMarquee.offsetHeight
+
+		if (parseInt(crossMarquee.style.top)>(actualHeight * (-1) + 8)) 
+		crossMarquee.style.top=parseInt(crossMarquee.style.top)-marqueeSpeed+"px" 
+		else 
+		crossMarquee.style.top=parseInt(marqueeheight)+8+"px"
+	},1000/30)
+}
+
+export {Init}
```
Comment (Albert221) [inline src/marquee.js]: If you'll make it `default export...` then this line is redundant.

File: src/tiles/FeedTile.vue [hunk 9]
```diff
@@ -1,15 +1,71 @@
...
+            }
+        },
+
+        mounted() {
+            this.updateNews()
+
+            setInterval(() => {
+                this.updateNews()
+            }, 1000 * 60 * 5); // Update news every 5 minutes
+            marquee.Init('marquee','vmarquee');
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Blank line before this line.

File: src/tiles/FeedTile.vue [hunk 10]
```diff
@@ -1,15 +1,71 @@
...
+                    response.json().then((data) => {
+                        this.feeds = data.news;
+                        
+                        this.convertDate();
+                    })
+                })
+            },
+
+            convertDate() {
+                for(var i=0;i<this.feeds.length;i++){
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Code style...

File: src/tiles/FeedTile.vue [hunk 11]
```diff
@@ -1,15 +1,71 @@
...
+                    this.feeds[i].publishedAt = new Date(1000*this.feeds[i].publishedAt);
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Code style again

File: src/tiles/FeedTile.vue [hunk 12]
```diff
@@ -20,14 +76,43 @@
...
+            height:100%;
+            overflow: hidden;
+        }
+
+        &--scroll {
+            position:absolute;
+        }
+
+        &--title {
+            font-size:1.1em;
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Space after property name

File: src/tiles/FeedTile.vue [hunk 13]
```diff
@@ -20,14 +76,43 @@
         flex-direction: column;
         padding: 1vw 1vw 0;
 
-        &--title {
+        &--advert {
             margin-top: 0;
         }
 
+        &--marquee {
+            position:relative;
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Space after property name

File: src/tiles/FeedTile.vue [hunk 14]
```diff
@@ -20,14 +76,43 @@
...
+            height:100%;
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Space after property name

File: src/tiles/FeedTile.vue [hunk 15]
```diff
@@ -20,14 +76,43 @@
...
+            overflow: hidden;
+        }
+
+        &--scroll {
+            position:absolute;
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Space after property name

File: src/tiles/FeedTile.vue [hunk 16]
```diff
@@ -20,14 +76,43 @@
...
+
         &--content {
             flex: 1;
             white-space: pre-line;
-            overflow: hidden;
+            text-align: center;
+        }
+
+        &--author {
+            text-align:right;
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Space after property name

File: src/tiles/FeedTile.vue [hunk 17]
```diff
@@ -20,14 +76,43 @@
...
+            font-style:italic;
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Space after property name

File: src/tiles/FeedTile.vue [hunk 18]
```diff
@@ -20,14 +76,43 @@
...
+            padding:0.5vw 1vw 0;
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Space after property name

File: src/tiles/FeedTile.vue [hunk 19]
```diff
@@ -20,14 +76,43 @@
...
+        }
+
+        &--publishedAt {
+            font-size:16px;
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Space after property name

File: src/tiles/FeedTile.vue [hunk 20]
```diff
@@ -20,14 +76,43 @@
...
+            text-align:right;
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Space after property name

File: src/tiles/FeedTile.vue [hunk 21]
```diff
@@ -20,14 +76,43 @@
...
+            padding:0 1vw 2vw;
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Space after property name

File: src/tiles/FeedTile.vue [hunk 22]
```diff
@@ -20,14 +76,43 @@
...
+            font-style:italic;
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Space after property name

File: src/tiles/FeedTile.vue [hunk 23]
```diff
@@ -1,15 +1,71 @@
...
 <script>
+    import * as marquee from './../marquee'
+    import fetch from 'fetch-retry'
+    import { API_URL } from '../constants'
+
     export default {
+        data() {
+            return {
+                feeds:[{
+                        "title": "",
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Single quotes

File: src/tiles/FeedTile.vue [hunk 24]
```diff
@@ -1,15 +1,71 @@
...
 
 <script>
+    import * as marquee from './../marquee'
+    import fetch from 'fetch-retry'
+    import { API_URL } from '../constants'
+
     export default {
+        data() {
+            return {
+                feeds:[{
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Space after property name

File: src/tiles/FeedTile.vue (hunk 9 above)
Comment (Albert221) [reply src/tiles/FeedTile.vue]: And IMO you should not pass this second parameter, but get child of first parameter

**Review by Albert221 — CHANGES_REQUESTED** [review]

File: src/marquee.js [hunk 25]
```diff
@@ -1,21 +1,18 @@
-var marqueeSpeed=1
-var actualHeight=0
+const marqueeSpeed=1
```
Comment (Albert221) [inline src/marquee.js]: Code style. Spaces surrounding operator

File: src/marquee.js [hunk 26]
```diff
@@ -1,21 +1,18 @@
...
+let actualHeight=0
 
-function Init(containerId, content)
-{
-	var crossMarquee=document.getElementById(content)
-	var marqueeheight=document.getElementById(containerId).offsetHeight
-	actualHeight=crossMarquee.offsetHeight
+export function Marquee(containerId) {
```
Comment (Albert221) [inline src/marquee.js]: `default`!

File: src/marquee.js [hunk 27]
```diff
@@ -1,21 +1,18 @@
...
+	var crossMarquee = document.getElementById(containerId).firstElementChild
+	var marqueeheight = document.getElementById(containerId).offsetHeight
```
Comment (Albert221) [inline src/marquee.js]: `var`

File: src/marquee.js [hunk 28]
```diff
@@ -1,21 +1,18 @@
...
+	actualHeight = crossMarquee.offsetHeight
 	
-	setInterval(function(){
-		var crossMarquee=document.getElementById(content)
-		actualHeight=crossMarquee.offsetHeight
+	setInterval(function() {
+		crossMarquee = document.getElementById(containerId).firstElementChild
+		actualHeight = crossMarquee.offsetHeight
 
 		if (parseInt(crossMarquee.style.top)>(actualHeight * (-1) + 8))
```
Comment (Albert221) [inline src/marquee.js]: Code style. Spaces surrounding operators! Redundant parenthesizes outside `-1`

File: src/marquee.js [hunk 29]
```diff
@@ -1,21 +1,18 @@
...
-	setInterval(function(){
-		var crossMarquee=document.getElementById(content)
-		actualHeight=crossMarquee.offsetHeight
+	setInterval(function() {
+		crossMarquee = document.getElementById(containerId).firstElementChild
+		actualHeight = crossMarquee.offsetHeight
 
 		if (parseInt(crossMarquee.style.top)>(actualHeight * (-1) + 8)) 
-		crossMarquee.style.top=parseInt(crossMarquee.style.top)-marqueeSpeed+"px" 
+			crossMarquee.style.top = parseInt(crossMarquee.style.top) - marqueeSpeed + 'px'
```
Comment (Albert221) [inline src/marquee.js]: IMO it would be better to use template string here instead of concatenation.

File: src/marquee.js [hunk 30]
```diff
@@ -1,21 +1,18 @@
...
+		actualHeight = crossMarquee.offsetHeight
 
 		if (parseInt(crossMarquee.style.top)>(actualHeight * (-1) + 8)) 
-		crossMarquee.style.top=parseInt(crossMarquee.style.top)-marqueeSpeed+"px" 
+			crossMarquee.style.top = parseInt(crossMarquee.style.top) - marqueeSpeed + 'px' 
 		else 
-		crossMarquee.style.top=parseInt(marqueeheight)+8+"px"
-	},1000/30)
+			crossMarquee.style.top = parseInt(marqueeheight) + 8 + 'px'
+	}, 1000/30)
```
Comment (Albert221) [inline src/marquee.js]: Code style. Spaces surrounding operator.

File: src/tiles/FeedTile.vue [hunk 31]
```diff
@@ -15,20 +15,20 @@
 </template>
 
 <script>
-    import * as marquee from './../marquee'
+    import { Marquee } from './../marquee'
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: Quick note: If you'll make `Marquee` a `default` export, then you'd want to remove this braces.

File: src/tiles/FeedTile.vue [hunk 32]
```diff
@@ -40,7 +40,8 @@
             setInterval(() => {
                 this.updateNews()
             }, 1000 * 60 * 5); // Update news every 5 minutes
-            marquee.Init('marquee','vmarquee');
+
+            { Marquee('marquee') };
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: What are those brackets for? o.O

File: src/tiles/FeedTile.vue [hunk 33]
```diff
@@ -61,8 +62,8 @@
             },
 
             convertDate() {
-                for(var i=0;i<this.feeds.length;i++){
-                    this.feeds[i].publishedAt = new Date(1000*this.feeds[i].publishedAt);
+                for(var i = 0; i < this.feeds.length; i++){
+                    this.feeds[i].publishedAt = new Date(1000 * this.feeds[i].publishedAt);
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: IMO you should use moment.js for dates because we use this in other parts of the frontend, but it's up to you!

**Review by Albert221 — CHANGES_REQUESTED** [review]

File: src/marquee.js [hunk 34]
```diff
@@ -0,0 +1,18 @@
+const marqueeSpeed = 1
+let actualHeight = 0
+
+export default function marquee(containerId) {
+	var crossMarquee = document.getElementById(containerId).firstElementChild
```
Comment (Albert221) [inline src/marquee.js]: **You still use `var`!!!**

File: src/tiles/FeedTile.vue [hunk 35]
```diff
@@ -1,15 +1,73 @@
...
-            Sed vestibulum suscipit efficitur. Quisque ac pretium urna, at ornare lorem. Proin metus felis, tristique a egestas et, aliquet at mauris. Morbi nec pellentesque ipsum. Nam consectetur justo id sollicitudin bibendum. Suspendisse potenti. Morbi mollis pharetra dolor, a lobortis sem. Nam quis varius neque. Proin imperdiet dignissim dui, a suscipit dui lobortis id. Integer iaculis id felis nec semper. Donec nisl magna, condimentum sed nisi in, dictum cursus ligula. Proin efficitur risus semper commodo sollicitudin. Phasellus ullamcorper mauris nec quam tempus sollicitudin.
+        <h2 class="feed--advert">Ogłoszenia</h2>
+        <div class="feed--marquee" id="marquee">
+            <div class="feed--scroll" id="vmarquee">
```
Comment (Albert221) [inline src/tiles/FeedTile.vue]: This id is redundant

**Review by Albert221 — CHANGES_REQUESTED** [review]
You almost made it. Two minor things to fix and it's done ;)

**Review by Albert221 — APPROVED** [review]

**Pull request merged**